app/livechat/server/api/v1/contact.ts

```diff
@@ -3,18 +3,19 @@ import { Meteor } from 'meteor/meteor';
 
 import { API } from '../../../../api/server';
 import { Contacts } from '../../lib/Contacts';
-import { LivechatVisitors } from '../../../../models';
+import { LivechatVisitors } from '../../../../models/server/index';
 
 API.v1.addRoute(
 	'omnichannel/contact',
 	{ authRequired: true },
 	{
-		post() {
+		async post() {
 			try {
 				check(this.bodyParams, {
 					_id: Match.Maybe(String),
 					token: String,
 					name: String,
+					username: String,
```

Comment:
🤔

Reply:
I added it since ```const contact = Contacts.registerContact(this.bodyParams);``` on line 24 would complain that this specific field is missing... :thinking

Reply:
Based on `Contacts.registerContact` method, this field seems optional